src/utils/parser/sources.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import type { Source, SourceId } from "debugger-html";
+
+var cachedSources = new Map();
+
+export function hasSource(sourceId: SourceId): boolean {
+  return cachedSources.has(sourceId);
+}
+
+export function setSource(source: Source) {
+  cachedSources.set(source.id, source);
+}
+
+export function getSource(sourceId: SourceId): Source {
+  if (!cachedSources.has(sourceId)) {
+    throw new Error(`${sourceId} was not provided.`);
+  }
+  // $FlowIgnore
```

Comment:
why are we ignoring here?

Reply:
Even if I did the check above, flow is convinced cacheSources.get can be undefined. Changing to ((x: any): Source) cast.